spacy/tokens/underscore.py

```diff
@@ -25,6 +25,9 @@ def __init__(
         obj: Union["Doc", "Span", "Token"],
         start: Optional[int] = None,
         end: Optional[int] = None,
+        label: Optional[Union[str, int]] = None,
+        kb_id: Optional[Union[str, int]] = None,
+        span_id: Optional[Union[str, int]] = None,
```

Comment:
I think this should only support `int` so that you don't have the situation where equivalent `str` and `int` representations look like different tuples. (I'm also on the fence about `Optional`. It might be easier/faster to have it be `0` in cases where it doesn't matter.)

Reply:
I've change it to `int = 0`